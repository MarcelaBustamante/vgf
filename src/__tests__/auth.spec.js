import React from 'react';
import {render, fireEvent} from '@testing-library/react';
import UserPanel from '../components/UserPanel';
import { UserProvider } from '../UserContext';
  
const mockedUsedNavigate = jest.fn();

  jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUsedNavigate,
  }));

  // Unit test for logout function
  describe('Logout', () => {
    beforeEach(() => {
      // fake storage
      localStorage.setItem('user', JSON.stringify({ email: 'user@example.com' }));
    });
  
    it('When user do log out then the storage should be clean', () => {
      // render user panel component
      const {getByText} = render(
        <UserProvider><UserPanel/></UserProvider>
      )
      // find log out button by text
      const button = getByText("Cerrar Sesi√≥n");
      fireEvent.click(button);
      // assert local storage is empty
      expect(localStorage.getItem('user')).toBeNull();
    });
  });
