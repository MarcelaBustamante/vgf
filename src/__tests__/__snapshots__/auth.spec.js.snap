// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout When user do log out then the storage should be clean 1`] = `
<div
  id="user-panel"
>
  <h2>
    Panel de Usuario
  </h2>
  <p>
    Email: 
    user@example.com
  </p>
  <button
    onClick={[Function]}
  >
    Editar Perfil
  </button>
  <button
    onClick={[Function]}
  >
    Cerrar Sesi√≥n
  </button>
</div>
`;
